packages/commerce-server/src/format-prices-for-product.ts

```diff
@@ -33,23 +34,52 @@ type FormatPricesForProductOptions = {
 }
 
 export async function getFixedDiscountForUpgrade({
-  upgradeProductId,
-  upgradeFromPurchase,
+  purchaseToBeUpgraded,
+  productToBePurchased,
   ctx = defaultContext,
 }: {
-  upgradeProductId: string
-  upgradeFromPurchase?: Purchase
+  purchaseToBeUpgraded: Purchase | null
+  productToBePurchased: Product
```

Comment:
I reworked the incoming params to make the naming more explicit and to make it so that we can compare the Purchase To Be Upgraded with the Product To Be Purchased.